tests/ai_pipeline/test_pipeline.py

```
@@ -0,0 +1,368 @@
+        assert len(masks) == 2
+
+    def test_no_exception_raised_during_full_flow(self, pipeline_instance, saved_test_image, dummy_bgr_image):
+        h, w = dummy_bgr_image.shape[:2]
+        pipeline_instance._mock_yolo.return_value = [
+            _make_yolo_result(np.array([[0, 0, w, h]], dtype=np.float32))
+        ]
+        pipeline_instance._mock_unet.return_value = _make_unet_output()
+
+        try:
+            pipeline_instance.run(str(saved_test_image))
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)